### PR TITLE
enh(servicegroup): add resource access rule configuration on cloud co…

### PR DESCRIPTION
**Pull request by adr-mo**
…ntext

Same behaviour than the hostgroup creation / edition https://github.com/centreon/centreon/pull/3814

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>
See Jira.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

## Summary by CodeRabbit

- **New Features**
	- Introduced a new filter for service groups to enhance user control and customization.
	- Added display of resource access rules in service group forms when applicable.
	- Updated help documentation to include guidance on `resource_access_rules`.

- **Refactor**
	- Overhauled service group database interaction functions for improved data management and ACL updates.
	- Enhanced code readability and maintainability in service group configurations by standardizing variable names and syntax.

### CODE REVIEW

File: centreon/www/include/configuration/configObject/servicegroup/DB-Func.php [hunk 1]
```diff
@@ -291,43 +291,248 @@ function multipleServiceGroupInDB($serviceGroups = [], $nbrDup = [])
...
+    return [];
+}
+
+function insertServiceGroupInDB(bool $isCloudPlatform = false, array $submittedValues = [])
+{
+    global $centreon, $form;
+
+    if ($submittedValues === []) {
+        $submittedValues = $form->getSubmitValues();
+    }
```
Comment (callapa) [inline centreon/www/include/configuration/configObject/servicegroup/DB-Func.php:467]: Faster.
```suggestion
    submittedValues = submittedValues ?: $form->getSubmitValues();
    }
```

File: centreon/www/include/configuration/configObject/servicegroup/formServiceGroup.php [hunk 2]
```diff
@@ -34,124 +34,143 @@
...
+    && $isCloudPlatform === true
+) {
+    $form->addElement(
+        'select2',
+        'resource_access_rules',
+        _('Resource Access Rule'),
+        [],
+        [
+            'datasourceOrigin' => 'ajax',
+            'availableDatasetRoute' => BASE_ROUTE . '?object=centreon_administration_aclgroup&action=list&for_cloud=true&all_servicegroups_filter=true',
```
Comment (callapa) [inline centreon/www/include/configuration/configObject/servicegroup/formServiceGroup.php:166]: Specifying the environment type in the request parameter is not secure.
```suggestion
            'availableDatasetRoute' => BASE_ROUTE . '?object=centreon_administration_aclgroup&action=list&all_servicegroups_filter=true',
```

Comment (adr-mo) [reply centreon/www/include/configuration/configObject/servicegroup/formServiceGroup.php]: Seen with callapa

File: centreon/www/include/configuration/configObject/servicegroup/DB-Func.php [hunk 3]
```diff
@@ -291,43 +291,248 @@ function multipleServiceGroupInDB($serviceGroups = [], $nbrDup = [])
...
+ * @param int $ruleId
+ * @param int $serviceGroupId
+ */
+function createNewDatasetFilter(int $datasetId, int $ruleId, int $serviceGroupId): void
+{
+    global $pearDB;
+
+    $query = <<<SQL
+        INSERT INTO dataset_filters (`type`, acl_resource_id, acl_group_id, resource_ids)
+        VALUES ('servicegroup', :datasetId, :ruleId, :serviceGroupId)
```
[the remark this answers is not in the document]
Comment (adr-mo) [reply centreon/www/include/configuration/configObject/servicegroup/DB-Func.php:385]: Yes later.

**Review by wtermellil — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**